proposals/release-versioning-testing-v2.md

```diff
@@ -0,0 +1,80 @@
+| C3            | v1     | v1  | v1           | v1         | v0                     | v0        | Simulates Router upgrading    |
+| C4            | v1     | v1  | v1           | v1         | v1                     | v0        | Simulates scheduler upgrading |
+| C5            | v1     | v1  | v1           | v1         | v1                     | v1        | Simulates cell upgrading      |
+
+## Notes<a id="sec-2-1" name="sec-2-1"></a>
+
+-   Deploy only one instance for all instance groups except Router and Cell to ensure routability to the test app at all times
+-   Cells are to be updated gracefuly (i.e. each cell evacuates before it stops) to ensure routability to the test app at all times
+-   Configurations follow the order of instance group updates as of [this version of cf-deployment](https://github.com/cloudfoundry/cf-deployment/commit/9be2644da8de08540891e24856bbdb88f9a83f67)
+-   We have to test both combinations of different versions of BBS and Locket since different versions can exist during a rolling update
+-   SSH proxy and auctioneer don't communicate with each other and exist on the same VM, hence we update them at the same time
+
+# Tests to run<a id="sec-3" name="sec-3"></a>
+
+## Long running tests<a id="sec-3-1" name="sec-3-1"></a>
+
+-   Similar to current [DUSTs](https://github.com/cloudfoundry/diego-upgrade-stability-tests) there will always be an app pushed right after `C0` and constantly checked for routability
+
+## Tests to run after each configuration update<a id="sec-3-2" name="sec-3-2"></a>
+
+-   Again, we can adapt DUSTs smoke test for this
+    -   push an app and make sure it is routable
+    -   scale up the app and make sure the 2 instances are routable
+-   Should also add ssh to the smoke test
+-   May be run a stripped down version of inigo/vizzini (doesn't have to be the actual test suite, as long as it offers the same test coverage) ?
+-   This could potentially be a separate Ginkgo `Context` each configuration which start the corresponding versions and run the smoke test
+
+# Questions<a id="sec-4" name="sec-4"></a>
+
+-   Is HTTP routing sufficient ?
+    -   we don't need to do anything since the routers upgrade before the cells (**preferred solution**)
+    -   it is the responsbility of the routing team to ensure the router/routing-api are backward compatible with an cells
+-   does everyone follow the same upgrade order in cf-dep ? What about cell blocks with different placement tags for example
+    -   We should communicate this requirement to operators, i.e. cells need to upgrade last (**preferred solution**)
+-   getting rid of bosh-lite removes all test coverage we had for bosh upgradability. for example not cleaning up pid files; which is something we ran into before
+    -   can we ask release-integration to own this ?
+        - not sure we should. It seems reasonable for us to own our own integration with bosh upgradability. The minimal test suite C0 -> C5 should be sufficient.
```

Comment:
@crhino had a concern that we aren't running any other bosh tests.
This is a real concern, however I think asking relint to own testing the upgradability of diego with bosh is unreasonable, given our product is a bosh-release. If we have concerns (such as the pid concern) about our upgradability with bosh, I think we should own that test, even if it needs to be an inigo-like suite and a separate small bosh-lite test.

Reply:
oops, we both tried to move the same convo for posterity.